python/lsst/afw/image/multiband.py

```diff
@@ -0,0 +1,1093 @@
+
+        Parameters
+        ----------
+        name: str
+            Name of the new mask plane
+        """
+        mask = self._singles[0]
+        idx = mask.addMaskPlane(name)
+        self._maskPlaneDict = mask.getMaskPlaneDict()
+        return idx
+
+    def removeMaskPlane(self, name):
+        """Remove a mask from the mask plane
+
+        Parameters
+        ----------
+        name: str
+            Name of the mask plane to remove
+        """
+        mask = self._singles[0]
+        mask.removeMaskPlane(name)
+        self._maskPlaneDict = mask.getMaskPlaneDict()
+
+    def clearAllMaskPlanes(self):
+        """Clear all the pixels
+        """
+        mask = self._singles[0]
+        mask.clearAllMaskPlanes()
+
+    def _getOtherMasks(self, others):
+        """Check if two masks can be combined
+
+        This method checks that `self` and `others`
+        have the same number of bands, or if
+        others is a single value, creates a list
+        to use for updating all of the `singles`.
+        """
+        if isinstance(others, MultibandMask):
+            if len(self.singles) != len(others.singles) or self.filters != others.filters:
+                msg = "Both `MultibandMask` objects must have the same number of bands to combine"
+                raise ValueError(msg)
+            result = [s for s in others.singles]
+        else:
+            result = [others]*len(self.singles)
+        return result
+
+    def __ior__(self, others):
+        _others = self._getOtherMasks(others)
+        singles = list(self.singles)
+        for n in range(len(self)):
+            singles[n] |= _others[n]
+        self._singles = tuple(singles)
+        return self
+
+    def __iand__(self, others):
+        _others = self._getOtherMasks(others)
+        singles = list(self.singles)
+        for n in range(len(self)):
+            singles[n] &= _others[n]
+        self._singles = tuple(singles)
+        return self
+
+    def __ixor__(self, others):
+        _others = self._getOtherMasks(others)
+        singles = list(self.singles)
+        for n in range(len(self)):
+            singles[n] ^= _others[n]
+        self._singles = tuple(singles)
+        return self
+
+    def __setitem__(self, args, value):
+        """Set a subset of the MultibandMask
+        """
+        if not isinstance(args, tuple):
+            indices = (args,)
+        else:
+            indices = args
+
+        # Return the single band object if the first
+        # index is not a list or slice.
+        filters, filterIndex = self.filterToIndex(indices[0])
+        if len(indices) > 1:
+            sy, sx = self.imageIndicesToNumpy(indices[1:])
+        else:
+            sy = sx = slice(None)
+        if isinstance(value, Mask):
+            self._array[filterIndex, sy, sx] = value.array
+        else:
+            self._array[filterIndex, sy, sx] = value
+
+    def set(self, value, filters=None, x=None, y=None):
```

Comment:
`set` is included to keep the same API as the single band `Image` and `Mask` classes, but I agree isn't needed if we implement `__setitem__`.

I wanted to include `__setitem__` because as an end user I'd like the image-like objects to be as close to duck typing numpy arrays as possible (this is one reason that I didn't like the `x,y` coordinate convention in `__getitem__`). So my preference would be to keep `__setitem__` and remove `set`.

Reply:
Sounds good.  `set` is actually being removed from the single-band objects on DM-14527.